backend/compact-connect/tests/unit/test_api.py

```diff
@@ -54,54 +55,25 @@ def _generate_expected_integration_object(self, handler_logical_id: str) -> dict
             }
         }
 
-    def test_synth_generates_provider_users_resource(self):
-        api_stack = self.app.sandbox_stage.api_stack
-        api_stack_template = Template.from_stack(api_stack)
-
-        # Ensure the resource is created with expected path
-        api_stack_template.has_resource_properties(
-            type=CfnResource.CFN_RESOURCE_TYPE_NAME,
-            props={
-                "ParentId": {
-                    # Verify the parent id matches the expected 'v1' resource
-                    'Ref': api_stack.get_logical_id(api_stack.api.v1_api.resource.node.default_child)
-                },
-                "PathPart": "provider-users"
-            })
-
-    def test_synth_generates_get_provider_users_me_endpoint_resource(self):
-        api_stack = self.app.sandbox_stage.api_stack
-        api_stack_template = Template.from_stack(api_stack)
-
-        # Ensure the resource is created with expected path
-        api_stack_template.has_resource_properties(
-            type=CfnResource.CFN_RESOURCE_TYPE_NAME,
-            props={
-                "ParentId": {
-                    # Verify the parent id matches the expected 'provider-users' resource
-                    'Ref': api_stack.get_logical_id(api_stack.api.v1_api.provider_users_resource.node.default_child)
-                },
-                "PathPart": "me"
-            })
-
-        # ensure the handler is created
-        api_stack_template.has_resource_properties(
-            type=CfnFunction.CFN_RESOURCE_TYPE_NAME,
-            props={
-                "Handler": "handlers.provider_users.get_provider_user_me"
-            })
-
-        # ensure the GET method is configured with the lambda integration and authorizer
-        api_stack_template.has_resource_properties(
-            type=CfnMethod.CFN_RESOURCE_TYPE_NAME,
-            props={
-                "HttpMethod": "GET",
-                # the provider users endpoints uses a separate authorizer from the staff endpoints
-                "AuthorizerId": {
-                    "Ref": api_stack.get_logical_id(api_stack.api.provider_users_authorizer.node.default_child)
-                },
-                # ensure the lambda integration is configured with the expected handler
-                "Integration": self._generate_expected_integration_object(
-                    api_stack.get_logical_id(
-                        api_stack.api.v1_api.provider_users.get_provider_users_me_handler.node.default_child)
-                )})
+    def compare_snapshot(self, actual: dict, snapshot_name: str, overwrite_snapshot: bool = False):
+        """
+        Compare the actual dictionary to the snapshot with the given name.
+        If overwrite_snapshot is True, overwrite the snapshot with the actual data.
+        """
+        snapshot_path = f'tests/resources/snapshots/{snapshot_name}.json'
```

Comment:
I don't think I've been 100% consistent on this, but we should probably `os.path.join(...)` these.